caddyconfig/httpcaddyfile/directives.go

```diff
@@ -119,6 +119,7 @@ type Helper struct {
 	matcherDefs  map[string]caddy.ModuleMap
 	parentBlock  caddyfile.ServerBlock
 	groupCounter counter
+	State        map[string]interface{}
```

Comment:
Add a godoc comment to this, maybe put it right after caddyfile.Dispenser near the top, because I personally like for exported fields to be at the top, but I don't care too much as far as merging this. Just needs godoc.